server/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQTerminationSubscriber.java

```diff
@@ -0,0 +1,140 @@
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+
+package org.apache.james.task.eventsourcing.distributed;
+
+import java.io.Closeable;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+
+import org.apache.james.backend.rabbitmq.ReactorRabbitMQChannelPool;
+import org.apache.james.backend.rabbitmq.SimpleConnectionPool;
+import org.apache.james.eventsourcing.Event;
+import org.apache.james.eventsourcing.eventstore.cassandra.JsonEventSerializer;
+import org.apache.james.lifecycle.api.Startable;
+import org.apache.james.task.eventsourcing.TerminationSubscriber;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.github.fge.lambdas.Throwing;
+import com.rabbitmq.client.AMQP;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.Delivery;
+import reactor.core.Disposable;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.WorkQueueProcessor;
+import reactor.core.scheduler.Schedulers;
+import reactor.rabbitmq.BindingSpecification;
+import reactor.rabbitmq.ExchangeSpecification;
+import reactor.rabbitmq.OutboundMessage;
+import reactor.rabbitmq.QueueSpecification;
+import reactor.rabbitmq.RabbitFlux;
+import reactor.rabbitmq.Receiver;
+import reactor.rabbitmq.ReceiverOptions;
+import reactor.rabbitmq.Sender;
+import reactor.rabbitmq.SenderOptions;
+
+public class RabbitMQTerminationSubscriber implements TerminationSubscriber, Startable, Closeable {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RabbitMQTerminationSubscriber.class);
+    private static final Integer MAX_CHANNELS_NUMBER = 1;
+    private static final String EXCHANGE_NAME = "terminationSubscriberExchange";
+    private static final String QUEUE_NAME_PREFIX = "terminationSubscriber";
+    private static final String ROUTING_KEY = "terminationSubscriberRoutingKey";
+
+    private final JsonEventSerializer serializer;
+    private final Mono<Connection> connectionMono;
+    private final ReactorRabbitMQChannelPool channelPool;
+    private final String queueName;
+    private WorkQueueProcessor<OutboundMessage> sendQueue;
+    private Flux<Delivery> listener;
+    private Disposable sendQueueHandle;
+
+    @Inject
+    public RabbitMQTerminationSubscriber(SimpleConnectionPool simpleConnectionPool, JsonEventSerializer serializer) {
+        this.serializer = serializer;
+        this.connectionMono = simpleConnectionPool.getResilientConnection();
+        this.channelPool = new ReactorRabbitMQChannelPool(connectionMono, MAX_CHANNELS_NUMBER);
+        this.queueName = QUEUE_NAME_PREFIX + UUID.randomUUID().toString();
```

Comment:
Can't we rather rely on the hostname as a queueName?

This will furthermore force a given James server to always rely on the same queue. Luckily this will result in James resumming its work after stopping, which sounds like a desirable property...

Reply:
not really

1. this is a non-durable queue, you will loose the `await`s anyway
2. why would you limit your ode to one queue?

Imagine a scenario where you have one queue per node:
If you have two consumers, it means that the two consumers won't have all the events.